presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -127,6 +130,52 @@ private static SubPlan analyzeGroupedExecution(Session session, Metadata metadat
         return new SubPlan(fragment, result.build());
     }
 
+    private SubPlan reassignPartitioningHandleIfNecessary(Session session, Metadata metadata, SubPlan subPlan)
+    {
+        return reassignPartitioningHandleIfNecessaryHelper(session, metadata, subPlan, subPlan.getFragment().getPartitioning());
+    }
+
+    private SubPlan reassignPartitioningHandleIfNecessaryHelper(Session session, Metadata metadata, SubPlan subPlan, PartitioningHandle newOutputPartitioningHandle)
```

Comment:
remove `Helper` part from the method name

Reply:
I prefer keeping the suffix as It is a helper method which is necessary in order to do recursion